lib/shopping_cart.rb

```diff
@@ -0,0 +1,28 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product_add)
+    @products << product_add
+  end
+
+  def details
+    cart = ShoppingCart.new("King Soopers", "30items")
+    cart_hash = cart.instance_variables.each_with_object({}) {|var, hash| hash[var[1..-2].to_sym] = cart.instance_variable_get(var) }
+    return cart_hash
```

Comment:
you should not be creating a new cart object -- you are currently in a cart object. If you wanted access to the cart you're in, you can use the keyword `self`.